dali/pipeline/data/tensor_vector.cc

```diff
@@ -1043,6 +1054,19 @@ void TensorVector<Backend>::ShareData(const shared_ptr<void> &ptr, size_t bytes,
 }
 
 
+template <typename Backend>
+void TensorVector<Backend>::setup_tensor_allocation(int index) {
+  if (tensors_[index].has_data() && tensors_[index].is_pinned() != is_pinned()) {
+    tensors_[index].Reset();
```

Comment:
Just to make sure, from here we will enter the second if and set the pinnedness, right?

Reply:
Yup, that is the intention. If we have data and the pinnedness is wrong, we Reset. 
Now we either are guaranteed to not have data or that pinnedness is consistent. We can set it only without data, so we do.

Reply:
Sounds good.